projects/rero/ng-core/src/lib/record/editor/wrappers/form-field-wrapper/form-field-wrapper.component.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * RERO angular core
+ * Copyright (C) 2020 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { FieldWrapper } from '@ngx-formly/core';
+import { EditorService } from '../../services/editor.service';
+
+@Component({
+  selector: 'ng-core-form-field-wrapper',
+  template: `
+    <div class="{{to.cssClass}} form-group" [class.has-error]="showError">
+      <!-- label -->
+      <label [attr.for]="id" *ngIf="to.label && to.hideLabel !== true" [tooltip]="to.description">
+        {{ to.label }}
+        <ng-container *ngIf="to.required && to.hideRequiredMarker !== true">&nbsp;*</ng-container>
+      </label>
+      <!-- clone button -->
+      <button type="button" *ngIf="canAdd()" (click)="add()" class="btn btn-link text-secondary btn-sm">
+        <i class="fa fa-clone"></i>
+      </button>
+      <!-- trash button -->
+      <button type="button" (click)="remove()" *ngIf="canRemove() && to.hideLabel !== true" class="btn btn-link text-secondary btn-sm">
+        <i class="fa fa-trash"></i>
+      </button>
+      <!-- field -->
+      <ng-template #fieldComponent></ng-template>
+      <div *ngIf="showError" class="invalid-feedback d-block">
+        <formly-validation-message [field]="field"></formly-validation-message>
+      </div>
+   </div>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class FormFieldWrapperComponent extends FieldWrapper {
+
+  constructor(private _editorService: EditorService) {
+    super();
+  }
+
+  /** Get the current field index position in the parent array.
+   *
+   * @returns number - the index position in the parent array, null if the parent
+   *                   is not an array.
+   */
+  getIndex() {
```

Comment:
If think this method is useless and you can do directly `Number(this.field.key)` instead of using it, because the check if field is array is already done